recipes/mysql/build.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# this script is based off the homebrew package:
+# https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb
+
+# Seems to be needed to build in some spots.
+for pth in /lib/cpp /lib64/cpp /usr/bin/cpp
+do
+    if [ ! -e ${pth} ]
+    then
+        # move along if these commands fail
+        mkdir -p `dirname ${pth}` || :
+        ln -s `which cpp` ${pth} || :
+    fi
+done
```

Comment:
This feels a little hacky/iffy to me. While it doesn't hurt the CIs, wouldn't feel comfortable running this on my own machine. Can you please point to where this is needed?

Reply:
So on the linux build CI, cmake cannot find cpp. See this job https://circleci.com/gh/conda-forge/staged-recipes/20853?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I have pasted the error here:

```
...
cd /staged-recipes/build_artefacts/mysql_1508768565518/work/mysql-5.7.20/build/rapid/plugin/group_replication/xdr_gen && rpcgen -C -h -o /staged-recipes/build_artefacts/mysql_1508768565518/work/mysql-5.7.20/build/rapid/plugin/group_replication/xdr_gen/xcom_vp.h xcom_vp.x
cannot find any C preprocessor (cpp)
rpcgen: C preprocessor failed with exit code 1
...
```

Reply:
So this came up in Arch Linux and they proposed a fix for [`rpcgen` calls]( https://bbs.archlinux.org/viewtopic.php?pid=1126049#p1126049 ). Guessing this doesn't happen on macOS.

Reply:
Ahhh cool! I will try that.

Reply:
Might try doing something like `CPP_ROOT="dirname $(which cpp)"` and then replacing the `rpcgen` calls to be something like `rpcgen -Y $CPP_ROOT`. Would need double quotes instead single quotes for `sed`.